pkg/deploy/agent.go

```diff
@@ -92,14 +94,15 @@ func (ac *AgentDeployConfig) AddImageFlag(imageTag string, flags *pflag.FlagSet)
 }
 
 func (ac *AgentDeployConfig) AddOptionFlags(flags *pflag.FlagSet) {
-	flags.DurationVar(&ac.DefaultPeriod, "default-period", 10*time.Second, "default period for jobs")
+	flags.DurationVar(&ac.DefaultPeriod, "default-period", 5*time.Second, "default period for jobs")
 	flags.BoolVar(&ac.DefaultSeccompProfileEnabled, "default-seccomp-profile", false, "if seccomp profile should be defaulted to RuntimeDefault for network-problem-detector pods")
 	flags.BoolVar(&ac.PingEnabled, "enable-ping", false, "if ICMP pings should be used in addition to TCP connection checks")
 	flags.BoolVar(&ac.PodSecurityPolicyEnabled, "enable-psp", true, "if pod security policy should be deployed")
 	flags.BoolVar(&ac.K8sExporterEnabled, "enable-k8s-exporter", false, "if node conditions and events should be updated/created")
 	flags.DurationVar(&ac.K8sExporterHeartbeat, "k8s-exporter-heartbeat", 3*time.Minute, "period for updating the node conditions by the K8s exporter")
 	flags.BoolVar(&ac.IgnoreAPIServerEndpoint, "ignore-gardener-kube-api-server", false, "if true, does not try to lookup kube api-server of Gardener control plane")
 	flags.StringVar(&ac.PriorityClassName, "priority-class", "", "priority class name")
+	flags.IntVar(&ac.MaxPeerNodes, "max-peer-nodes", 0, "if != 0 restricts number of peer nodes used as check destinations")
```

Comment:
Do we want to provide a sensible default here or should it be configured in or via the corresponding gardener extension?

Reply:
If you are using the network problem detector directly, there should be no problem with the number of metrics.
Therefore I would leave the default to check all nodes in this case.
The gardener extension provides its own configuration anyway, so `maxPeerNodes` would be enabled in the extension by default.